### PR TITLE
feat(server): add REST API to delete work pool's worker

### PR DESCRIPTION
**Pull request by hainenber**
Partially implements #12176 

Add a REST API to delete work pool's workers. This should serve as baseline to allow deleting workers using API, which will be implemented in `prefect-ui-library`.

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Comment by znicholasbrown** [comment]
Hi @hainenber - thanks for tackling this! We'd like to make sure this behavior is matched in our own backend so if you wouldn't mind I'm going to hold off on approving/merging this until Thursday morning, which will ensure it gets in the next release 😄 

The same will go for your corresponding UI PR - https://github.com/PrefectHQ/prefect-ui-library/pull/2231; once this is merged on Thursday we can merge and release that one as well. Thanks again, this looks great!

File: src/prefect/server/api/workers.py [hunk 1]
```diff
@@ -599,3 +599,27 @@ async def read_workers(
...
+    Delete a work pool's worker
+    """
+
+    async with db.session_context(begin_transaction=True) as session:
+        work_pool_id = await worker_lookups._get_work_pool_id_from_name(
+            session=session, work_pool_name=work_pool_name
+        )
+        await models.workers.delete_worker(
+            session=session, work_pool_id=work_pool_id, worker_name=worker_name, db=db
+        )
```
Comment (jakekaplan) [inline src/prefect/server/api/workers.py:625]: Would you mind updating this so it returns 404 if nothing is deleted?

```suggestion
        deleted = await models.workers.delete_worker(
            session=session, work_pool_id=work_pool_id, worker_name=worker_name
        )
        if not deleted:
            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Worker not found.")

```

File: tests/server/api/test_workers.py [hunk 2]
```diff
@@ -849,6 +849,47 @@ async def test_worker_status_accounts_for_heartbeat_interval(
...
+        insert_stmt = (await db.insert(db.Worker)).values(
+            name=worker_name,
+            work_pool_id=work_pool_id,
+            last_heartbeat_time=pendulum.now(),
+        )
+        await session.execute(insert_stmt)
+        await session.commit()
+
+        response = await client.delete(
+            f"/work_pools/${work_pool.name}/workers/does-not-exist"
```
Comment (jakekaplan) [inline tests/server/api/test_workers.py:888]: Not sure if this intentional but this is checking for a work pool that doesn't exist with `${work_pool.name}`. this is still a good test!

Could another one that a valid work pool but non-existent worker gives 404?

**Comment by znicholasbrown** [comment]
@hainenber apologies - we'll need to make sure comments are addressed before merging; that will likely mean we miss today's release

File: tests/server/api/test_workers.py (hunk 2 above)
Comment (hainenber) [reply tests/server/api/test_workers.py]: I've fixed this by creating a new valid workpool. PTAL in latest commit :D

**Comment by hainenber** [comment]
Apologies, I was a bit busy lately and not able to catch up with GH messages. I've addressed the comments now

**Review by jakekaplan — APPROVED** [review]

File: tests/server/api/test_workers.py [hunk 3]
```diff
@@ -849,6 +849,48 @@ async def test_worker_status_accounts_for_heartbeat_interval(
...
+        )
+        assert deleted_worker_name not in map(lambda x: x.name, remaining_workers)
+
+    async def test_nonexistent_worker(self, client, session, db):
+        worker_name = "worker1"
+        wp = await models.workers.create_work_pool(
+            session=session,
+            work_pool=schemas.actions.WorkPoolCreate(name="A"),
+        )
+        insert_stmt = (await db.insert(db.Worker)).values(
```
Comment (jakekaplan) [inline tests/server/api/test_workers.py:882]:
```suggestion
        insert_stmt = (db.insert(db.Worker)).values(
```

File: tests/server/api/test_workers.py [hunk 4]
```diff
@@ -849,6 +849,48 @@ async def test_worker_status_accounts_for_heartbeat_interval(
...
+            name=worker_name,
+            work_pool_id=wp.id,
+            last_heartbeat_time=pendulum.now(),
+        )
+        await session.execute(insert_stmt)
+        await session.commit()
+
+        response = await client.delete(f"/work_pools/${wp.name}/workers/does-not-exist")
```
Comment (jakekaplan) [inline tests/server/api/test_workers.py:890]:
```suggestion
        response = await client.delete(f"/work_pools/{wp.name}/workers/does-not-exist")
```

File: tests/server/api/test_workers.py [hunk 5]
```diff
@@ -849,6 +849,48 @@ async def test_worker_status_accounts_for_heartbeat_interval(
         assert workers_response.json()[0]["status"] == "OFFLINE"
 
 
+class TestDeleteWorker:
+    async def test_delete_worker(self, client, work_pool, session, db):
+        work_pool_id = work_pool.id
+        deleted_worker_name = "worker1"
+        for i in range(2):
+            insert_stmt = (await db.insert(db.Worker)).values(
```
Comment (jakekaplan) [inline tests/server/api/test_workers.py:857]:
```suggestion
            insert_stmt = (db.insert(db.Worker)).values(
```

**Comment by jakekaplan** [comment]
@hainenber thanks for updating! I missed those on my first look through. Will re-run the tests now

**Pull request merged**